scripts/config.py

```diff
@@ -166,6 +166,7 @@ def include_in_full(name):
     if name in [
             'MBEDTLS_CTR_DRBG_USE_128_BIT_KEY',
             'MBEDTLS_DEPRECATED_REMOVED',
+            'MBEDTLS_DEPRECATED_WARNING',
```

Comment:
We have a backlog of writing comments to explain this list. A long time ago I wrote some for `config.pl` but that never got merged.

Reply:
Well I think it adjusting the definition of full is the best time to finally add those comments. Otherwise this PR will only fix one half of the problem (the definition was suboptimal) but not the other half (different team members at different times had different enough conceptions of what "full" meant that we broke things), which is IMO at least as important.